### PR TITLE
Upgrade and cleanup dependencies

### PR DESCRIPTION
**Pull request by mikeharder**
- Remove unnecessary dependencies (many already transitive deps)
- In `rulesets`, move some deps from dev to runtime
  - May have worked in practice as dev deps, but was technically incorrect
- Upgrade all dependencies to latest major, except for these which require reactions to breaking changes:
  - `typescript`
  - `@apidevtools/json-schema-ref-parser`
  - `vscode-jsonrpc`
  - `rollup`, `@rollup/plugin-commonjs`, `@rollup/plugin-json`
  - `jest`, `ts-jest`, `@types/jest`

### CODE REVIEW

File: packages/azure-openapi-validator/autorest/package.json [hunk 1]
```diff
@@ -49,22 +49,11 @@
...
     "@stoplight/json-ref-resolver": "^3.1.6",
-    "@stoplight/better-ajv-errors": "1.0.1",
-    "ajv": "^8.10.0",
-    "@stoplight/json": "~3.17.1",
-    "@stoplight/lifecycle": "2.3.2",
-    "@stoplight/path": "1.3.2",
-    "@stoplight/spectral-parsers": "^1.0.0",
-    "@stoplight/spectral-ref-resolver": "^1.0.0",
-    "@stoplight/types": "^13.5.0",
+    "@stoplight/types": "^14.1.1",
```
Comment (mikeharder) [inline packages/azure-openapi-validator/autorest/package.json:56]: Removed deps were already transitive deps that didn't need to be direct

**Review by konrad-jamrozik — APPROVED** [review]

File: packages/azure-openapi-validator/autorest/src/spectral-plugin-func.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import {
 import { Resolver } from "@stoplight/json-ref-resolver"
 import { IResolver } from "@stoplight/json-ref-resolver/types"
 import { Ruleset, Spectral } from "@stoplight/spectral-core"
-import { safeLoad } from "js-yaml"
+import { load } from "js-yaml"
```
Comment (mikeharder) [inline packages/azure-openapi-validator/autorest/src/spectral-plugin-func.ts:14]: Breaking change in `js-yaml@4`

File: packages/rulesets/src/native/tests/utilities/tests-helper.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 import { LintResultMessage, OpenApiTypes, LintTester, IRuleSet, RulesObject } from "@microsoft.azure/openapi-validator-core"
-import { safeLoad } from "js-yaml"
+import { load } from "js-yaml"
```
Comment (mikeharder) [inline packages/rulesets/src/native/tests/utilities/tests-helper.ts:6]: Breaking change in `js-yaml@4`

File: packages/rulesets/rollup.config.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ export default [
     exports: 'default'
   },
   plugins: [commonjs(),json()],
-  external:["@stoplight/spectral-functions","@stoplight/spectral-formats","@stoplight/spectral-rulesets"]
+  external:["@stoplight/spectral-functions","@stoplight/spectral-formats"]
```
Comment (mikeharder) [inline packages/rulesets/rollup.config.js:12]: Removed dep from package.json

**Pull request merged**